File: Data Structures & Algorithms/top-k-elements-in-list/submission-11.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        List<Integer> [] frequency = new List[nums.length + 1];
        Map<Integer, Integer> count = new HashMap<>();

        for(int i = 0; i < frequency.length; i++){
            frequency[i] = new ArrayList<>();
        }

        for(int num: nums){
            count.put(num, count.getOrDefault(num, 0) + 1);
        }

        for(Map.Entry<Integer, Integer> current: count.entrySet()){
            System.out.println("Key: " + current.getKey() + " Value: " + current.getValue());
            frequency[current.getValue()].add(current.getKey());
        }

        int [] result = new int [k];
        int index = 0;

        for(int i = frequency.length - 1; i > 0; i--){
            for(int num: frequency[i]){
                result[index++] = num;

                if(index == k) return result;
            }

        }


        return result;
    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-11.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        current_length = len(nums)
        left_array = [0] * current_length
        right_array = [0] * current_length
        result = [0] * current_length

        value = 1
        for index, num in enumerate(nums):
            left_array[index] = value
            value *= num
        
        value = 1

        for index in range(len(nums) -1, -1, -1):
            right_array[index] = value
            value *= nums[index]
        
        for index in range(current_length):
            result[index] = right_array[index] * left_array[index]


        return result
